### PR TITLE
Change format of display address

### PR DESCRIPTION
**Pull request by benjefferies**
# Motivation and Context
Change the display address to contain 'district', 'post_town',
'postcode' if present and 2 fields haven't already been populated. Also
comma separating

# What has changed
Add 'district', 'post_town', 'postcode' to display address
Comma separating address

# How to test?
Run unit tests

# Links
https://trello.com/c/DKeaOOEE/178-send-address-and-country-code-to-eq-from-rh-5

# Screenshots (if appropriate):

### CODE REVIEW

File: app/eq.py [hunk 1]
```diff
@@ -195,22 +195,21 @@ def camel_to_snake(s):
     @staticmethod
     def build_display_address(sample_attributes):
         """
-        Build `display_address` value by appending not-None (in order) prem values of sample attributes
+        Build `display_address` value by appending not-None (in order) values of sample attributes
 
-        :param sample_attributes: dictionary of attributes with Prem1, Prem2, Prem3, Prem4 keys present
+        :param sample_attributes: dictionary of address attributes
         :return: string of a single prem value or a combination of two
```
Comment (jcox-dev) [inline app/eq.py]: Might be worth changing this to something like:

`:return: string of a single address attribute or a combination of two`

File: app/eq.py [hunk 2]
```diff
@@ -195,22 +195,21 @@ def camel_to_snake(s):
...
         """
         display_address = ''
-        for prem_key in ['Prem1', 'Prem2', 'Prem3', 'Prem4']:  # retain order of Prem values
-            try:
-                val = sample_attributes[prem_key]
-            except KeyError:
-                raise InvalidEqPayLoad(f"{prem_key} missing from sample attributes")
+        for prem_key in ['Prem1', 'Prem2', 'Prem3', 'Prem4', 'District', 'PostTown', 'Postcode']:  # retain order of Prem values
```
Comment (jcox-dev) [inline app/eq.py]: Suggested change to comment:

`# retain order of address attributes`

File: app/eq.py [hunk 3]
```diff
@@ -195,22 +195,21 @@ def camel_to_snake(s):
...
+            val = sample_attributes.get(prem_key)
             if val:
                 prev_display = display_address
-                display_address = f'{prev_display} {val}' if prev_display else val
+                display_address = f'{prev_display}, {val}' if prev_display else val
                 if prev_display:
                     break  # break once two prems have been added
```
Comment (jcox-dev) [inline app/eq.py]: `# break once two address attributes have been added`

File: tests/unit/test_eq.py [hunk 4]
```diff
@@ -290,12 +290,10 @@ def test_build_display_address(self):
...
         from app import eq
 
-        attributes = self.sample_attributes_json['attributes'].copy()
-        del attributes['Prem2']
+        attributes = {}
 
-        with self.assertRaises(InvalidEqPayLoad) as ex:
+        with self.assertRaises(InvalidEqPayLoad):
             eq.EqPayloadConstructor.build_display_address(attributes)
-        self.assertIn("Prem2 missing from sample attributes", ex.exception.message)
```
Comment (jcox-dev) [inline tests/unit/test_eq.py]: Could add this back in as:

`self.assertIn("Displayable address not in sample attributes", ex.exception.message)`

File: tests/unit/test_eq.py [hunk 5]
```diff
@@ -330,55 +328,82 @@ def test_build_display_address_prems(self):
...
+             }, "Another House, In Brizzle"),
+            ({
+                 "Prem1": "",
+                 "Prem2": "",
+                 "Prem3": "",
+                 "Prem4": "Another House",
+                 "District": "In The Shire",
+                 "PostTown": "",
+                 "Postcode": ""
+             }, "Another House, In The Shire"),
```
Comment (jcox-dev) [inline tests/unit/test_eq.py]: 👍

File: tests/unit/test_eq.py (hunk 4 above)
Comment (benjefferies) [reply tests/unit/test_eq.py]: Done

File: app/eq.py (hunk 3 above)
Comment (benjefferies) [reply app/eq.py]: Done

File: app/eq.py (hunk 2 above)
Comment (benjefferies) [reply app/eq.py]: Done

File: app/eq.py (hunk 1 above)
Comment (benjefferies) [reply app/eq.py]: Done

**Comment by jcox-dev** [comment]
[Send address and country code to eQ from RH (5)](https://trello.com/c/DKeaOOEE/178-send-address-and-country-code-to-eq-from-rh-5)

**Pull request merged**